include/revng/ADT/TypeTraits.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <type_traits>
+
+template<typename T>
+concept NotVoid = not std::is_void_v<T>;
+
+template<class Derived, class Base>
+concept DerivesFrom = std::is_base_of_v<Base, Derived>;
```

Comment:
This seems to belong to `Concepts.h`, see if we can avoid a new header.

Reply:
Done. Also moved `Concepts.h` to `ADT` since it belongs there to be reusable.